### PR TITLE
fix problems in class Address

### PR DESCRIPTION
**Pull request by kurtkeller**
- name should be 1-70 chars, which is 0 < X < 71,
  rather than 1 < X < 70

- The info in the QR-code image itself contained
  None instead of the empty string for street
  if no value was passed at __init__.
  Somehow this does not show up in the tests which
  check bill.qr_data(), but it does when decoding
  the generated QR-code image.

- Changed the code in __init__ do use the same
  initialization steps for each parameter because:
  - some changes (at least for street) were required
  - strip()ing the passed parameters should be done
     if not X   does not the right thing if   X=' '
  - handling was not consistent
    (X or '' / X and len(X) / if not X)

### CODE REVIEW

File: qrbill/bill.py [hunk 1]
```diff
@@ -51,52 +51,58 @@
 
 
 class Address:
-    def __init__(self, *, name='', street='', house_num='', pcode=None, city=None, country=None):
-        if not (1 < len(name or '') < 70):
+    def __init__(self, *, name=None, street=None, house_num=None, pcode=None, city=None, country=None):
+        self.name = name or ''
```
Comment (claudep) [inline qrbill/bill.py]: Could you explain why you prefer setting `None` as default value then add this line to set a default?

Comment (claudep) [reply qrbill/bill.py]: OK, I guess it's your `consistent` argument is your commit description.

Comment (kurtkeller) [reply qrbill/bill.py]: If, for example, street is not specified on the command line, then we get a None value passed to __init__ anyway from the caller, due to argparse not having a value for the argument. Keeping that as None can cause problems in the QR-code.

This could also be used as a library in another program, without using scripts/qrbill. Where does the data from such another program come from? If there is, let's say no street and house number in the customer DB of that program, will it pass None or an empty string? We don't know.

So, basically, in the processing code we have to make sure we convert input to sensible values for each of the passed parameters anyway. In that case I think it is better to be consistent in what the defaults in __init__ are (we could also agree on '') and have some kind of common processing-path to follow in the code for each parameter. That makes it easier to understand and easier to extend in the future, should that ever be necessary.

**Comment by claudep** [comment]
Thanks a lot, pushed in  c51c5e78d330d1c and a7dd10b6abb8b518
